geoext.ux/ux/FeatureEditing/lib/GeoExt.ux/FeatureEditorGrid.js

```diff
@@ -318,11 +341,77 @@ GeoExt.ux.FeatureEditorGrid = Ext.extend(Ext.grid.EditorGridPanel, {
      */
     onFeaturemodified: function(e) {
         if(e.feature === this.store.feature) {
-            this.dirty = true;
-            this.saveButton.setDisabled();
+            if (e.action == 'undo') {
+                // Do nothing
+            }
+            else {
+                // Don't stash twice on first modification
+                if (!!e.feature.state) {
+                    this.stash(e.feature);
+                    this.dirty = true;
+                }
+            }
+            this.saveButton.setDisabled(!this.dirty);
         }
     },
 
+    /** private: method[stash]
+     *  :param feature: ``OpenLayers.Feature`` The feature to stash.
+     *
+     *  Stash feature state and properties for the undo functionality.
+     */
+    stash: function(feature) {
+        if (!!this.previous) {
```

Comment:
`this.previous !== null` is relay better!, and it should be initialized to to null and documented